src/main/java/com/gtnewhorizon/randomboubles/research/RingResearch.java

```diff
@@ -0,0 +1,702 @@
+            -4,
+            0,
+            new ItemStack(BaubleItems.combinationRing, 1, 1)).setParents("RBSPELLCASTERSRINGOFTHESKY")
+                .setPages(pages)
+                .registerResearchItem();
+        // Exotic Ring of the Earth
+        pages = new ResearchPage[] { new ResearchPage("randomboubles_research_page.RBEXOTICRINGOFTHEEARTH.1"),
+            new ResearchPage((InfusionRecipe) recipeList.get("RBExoticRingoftheEarth")) };
+        researchAspects = new AspectList().add(Aspect.EARTH, 32)
+            .add(Aspect.WATER, 32)
+            .add(Aspect.ENTROPY, 32)
+            .add(Aspect.MAGIC, 64)
+            .add(Aspect.AURA, 64);
+        getResearchItem(
+            "RBEXOTICRINGOFTHEEARTH",
+            "RBRINGS",
+            researchAspects,
+            -6,
+            0,
+            0,
+            new ItemStack(BaubleItems.combinationRing, 1, 4)).setParents("RBSPELLSLINGERSRINGOFTHEEARTH")
+                .setPages(pages)
+                .registerResearchItem();
+        // Runeforged Ring of the Sky
+        pages = new ResearchPage[] { new ResearchPage("randomboubles_research_page.RBRUNEFORGEDRINGOFTHESKY.1"),
+            new ResearchPage((InfusionRecipe) recipeList.get("RBRuneforgedRingOfTheSky")) };
+        researchAspects = new AspectList().add(Aspect.AIR, 64)
+            .add(Aspect.FIRE, 64)
+            .add(Aspect.ORDER, 64)
+            .add(Aspect.MAGIC, 64)
+            .add(Aspect.AURA, 64);
+        getResearchItem(
+            "RBRUNEFORGEDRINGOFTHESKY",
+            "RBRINGS",
+            researchAspects,
+            -7,
+            -5,
+            0,
+            new ItemStack(BaubleItems.combinationRing, 1, 2)).setParents("RBEXOTICRINGOFTHESKY")
+                .setPages(pages)
+                .registerResearchItem();
+        // Runeforged Ring of the Earth
+        pages = new ResearchPage[] { new ResearchPage("randomboubles_research_page.RBRUNEFORGEDRINGOFTHEEARTH.1"),
+            new ResearchPage((InfusionRecipe) recipeList.get("RBRuneforgedRingoftheEarth")) };
+        researchAspects = new AspectList().add(Aspect.EARTH, 64)
+            .add(Aspect.WATER, 64)
+            .add(Aspect.ENTROPY, 64)
+            .add(Aspect.MAGIC, 64)
+            .add(Aspect.AURA, 64);
+        getResearchItem(
+            "RBRUNEFORGEDRINGOFTHEEARTH",
+            "RBRINGS",
+            researchAspects,
+            -7,
+            1,
+            0,
+            new ItemStack(BaubleItems.combinationRing, 1, 5)).setParents("RBEXOTICRINGOFTHEEARTH")
+                .setPages(pages)
+                .registerResearchItem();
+
+        // Thaumaturge's Ring
+        pages = new ResearchPage[] { new ResearchPage("randomboubles_research_page.RBTHAUMATURGESRING.1"),
+            new ResearchPage((InfusionRecipe) recipeList.get("RBThaumaturgesRing")) };
+        researchAspects = new AspectList().add(Aspect.EARTH, 64)
+            .add(Aspect.AIR, 64)
+            .add(Aspect.EARTH, 64)
```

Comment:
Applies to all Thaumaturge's rings. Also see my comment on RBTHEORYRINGOFMAGICALPROTECTION about increasing the complexity for more space and shifting the order of aspects for variety.
```suggestion
            .add(Aspect.FIRE, 64)
```